packages/search/src/search.tsx

```diff
@@ -0,0 +1,417 @@
+
+	clear = () => this.setState( { keyword: '' } );
+
+	onBlur = ( event: MouseEvent< HTMLInputElement > ) => {
+		if ( this.props.onBlur ) {
+			this.props.onBlur( event );
+		}
+
+		this.setState( { hasFocus: false } );
+	};
+
+	onChange = ( keyword: string ) => {
+		this.setState( {
+			keyword,
+		} );
+	};
+
+	keyUp = ( event: KeyboardEvent< HTMLInputElement > ) => {
+		if ( event.key === 'Enter' && isMobile() ) {
+			//dismiss soft keyboards
+			this.blur();
+		}
+
+		if ( ! this.props.pinned ) {
+			return;
+		}
+
+		if ( event.key === 'Escape' ) {
+			this.closeListener( event );
+		}
+		this.scrollOverlay();
+	};
+
+	keyDown = ( event: KeyboardEvent< HTMLInputElement > ) => {
+		this.scrollOverlay();
+
+		// @todo(saramarcondes) investigate why `target` doesn't have the expected type
+		// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+		// @ts-ignore
+		if ( event.key === 'Escape' && event.target?.value === '' ) {
+			this.closeListener( event );
+		}
+
+		this.props.onKeyDown?.( event );
+	};
+
+	// Puts the cursor at end of the text when starting
+	// with `initialValue` set.
+	onFocus = () => {
+		if ( ! this.searchInput.current ) {
+			return;
+		}
+
+		const setValue = this.searchInput.current.value ?? '';
+		if ( setValue ) {
+			// Firefox needs clear or won't move cursor to end
+			this.searchInput.current.value = '';
+			this.searchInput.current.value = setValue;
+		}
+
+		this.setState( { hasFocus: true } );
+		this.props.onSearchOpen();
+	};
+
+	render() {
+		const searchValue = this.state.keyword;
+		const placeholder = this.props.placeholder || __( 'Search…' );
+		const inputLabel = this.props.inputLabel;
+		const enableOpenIcon = this.props.pinned && ! this.state.isOpen;
+		const isOpenUnpinnedOrQueried =
+			this.state.isOpen || ! this.props.pinned || this.props.initialValue;
+
+		const autocorrect = this.props.disableAutocorrect && {
+			autoComplete: 'off',
+			autoCorrect: 'off',
+			spellCheck: 'false',
+		};
+
+		const searchClass = classNames( this.props.className, this.props.dir, {
+			'is-expanded-to-container': this.props.fitsContainer,
+			'is-open': isOpenUnpinnedOrQueried,
+			'is-searching': this.props.searching,
+			'is-compact': this.props.compact,
+			'has-focus': this.state.hasFocus,
+			'has-open-icon': ! this.props.hideOpenIcon,
+			search: true,
+		} );
+
+		const fadeDivClass = classNames( 'search__input-fade', this.props.dir );
+		const inputClass = classNames( 'search__input', this.props.dir );
+
+		return (
+			<div dir={ this.props.dir } className={ searchClass } role="search">
+				<Spinner />
+				<div
+					role="button"
+					className="search__icon-navigation"
+					ref={ this.openIcon }
+					onClick={ enableOpenIcon ? this.openSearch : this.focus }
+					tabIndex={ enableOpenIcon ? 0 : undefined }
+					onKeyDown={ enableOpenIcon ? this.openListener : undefined }
+					aria-controls={ 'search-component-' + this.instanceId }
+					aria-label={ __( 'Open Search' ) }
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **31** times:
`translate( 'Open Search', { context: 'button label'} )` *ES Score: 14*